### PR TITLE
Implement available_plugins, client_started, and config_changed notifications.

### PR DESCRIPTION
**Pull request by bytebuddha**
Just a few additions i added to play around with some newer xi commit's. Feel free to close this if your already working on it, or wish to go in a different direction.

### CODE REVIEW

File: examples/it_works.rs [hunk 1]
```diff
@@ -5,7 +5,7 @@ extern crate xrl;
 
 use futures::{future, Future, Stream};
 use tokio_core::reactor::Core;
-use xrl::{AvailablePlugins, ConfigChanged, spawn, Client, Frontend, FrontendBuilder, ScrollTo, ServerResult, Style, Update};
```
Comment (little-dude) [inline examples/it_works.rs]: I know it's a bit ugly here, but I really like having stuff imported explicitely.

**Comment by bytebuddha** [comment]
alright @little-dude should be ready now. Just for the record this only implement's receiving config updates from core not actually changing them. In case you missed it they added a doc [here](https://github.com/google/xi-editor/blob/master/doc/config.md)

File: src/frontend.rs [hunk 2]
```diff
@@ -2,7 +2,7 @@ use errors::ServerError;
 use protocol::Service;
 use futures::{future, Future};
 use serde_json::{from_value, Value};
-use structs::{ScrollTo, Style, Update};
```
Comment (little-dude) [inline src/frontend.rs]: You forgot this one in your last commit :)

Comment (bytebuddha) [reply src/frontend.rs]: Oop's. it is fixed, sorry about that.

**Comment by little-dude** [comment]
Thanks @bytebuddha !
Do you want a release with these changes?

**Comment by bytebuddha** [comment]
No not quite yet, there are a few other PR's i wanted to make and i was planning to write another example an 'repl' type thing to debug xi-core.

**Comment by little-dude** [comment]
awesome!

**Pull request merged**